Packages/com.unity.inputsystem/InputSystem/Plugins/UI/InputSystemUIInputModule.cs

```diff
@@ -1845,6 +1865,11 @@ private int AllocatePointer(int pointerId, int touchId, UIPointerType pointerTyp
                 eventData = new ExtendedPointerEventData(eventSystem);
 
             eventData.pointerId = pointerId;
+
+#if UNITY_INPUT_SYSTEM_ENABLE_MULTIDISPLAY && (UNITY_EDITOR_WIN || UNITY_EDITOR_OSX || UNITY_EDITOR_LINUX || UNITY_ANDROID)
+            eventData.displayIndex = displayIndex;
```

Comment:
Isn't this range assignment a cast from int to byte? Maybe I am wrong but I noticed there is both byte and int representations of the displayIndex so it would be good to debug-assert any occurances of truncation with a debug-only range check to detect any incorrect out-of-range effects

Reply:
I don't have a good understanding for the flow of execution through this codebase.  I added:
```
Debug.Assert(m_CurrentDisplayIndices[pointerIndex] <= byte.MaxValue, "Display index was larger than expected");
```

To TouchSimluation based on a previous comment on this PR (which I can't find now).  It's TouchSimulation:294.  

I'm happy to add more of those, but I have no idea if they're superfluous or not.  Do you know?

Reply:
Added :)

Reply:
Yes, on the managed side the displayIndex is a byte and when it came to blitting from unmanaged to managed code, I did make an attempt to replace with a ByteControl instead of IntegerControl, but either one did not exist, or it broke the blit, I do not remember which